Framework/HistogramData/test/HistogramRebinTest.h

```diff
@@ -0,0 +1,324 @@
+#ifndef MANTID_HISTOGRAMDATA_HISTOGRAMREBINTEST_H_
+#define MANTID_HISTOGRAMDATA_HISTOGRAMREBINTEST_H_
+
+#include <cxxtest/TestSuite.h>
+
+#include "MantidHistogramData/Histogram.h"
+#include "MantidHistogramData/HistogramRebin.h"
+#include "MantidHistogramData/LinearGenerator.h"
+
+#include <algorithm>
+#include <random>
+
+using namespace Mantid::HistogramData;
+
+class HistogramRebinTest : public CxxTest::TestSuite {
+public:
+  // This pair of boilerplate methods prevent the suite being created statically
+  // This means the constructor isn't called when running other tests
+  static HistogramRebinTest *createSuite() { return new HistogramRebinTest(); }
+  static void destroySuite(HistogramRebinTest *suite) { delete suite; }
+
+  void testExecRebinCounts() {
+    TS_ASSERT_THROWS_NOTHING(rebinCounts(
+        getCountsHistogram(), BinEdges(10, LinearGenerator(0, 0.5))));
+  }
+
+  void testExecRebinFrequency() {
+    TS_ASSERT_THROWS_NOTHING(rebinFrequencies(
+        getFrequencyHistogram(), BinEdges(10, LinearGenerator(0, 0.5))));
+  }
+
+  void testRebinFrequenciesFailModes() {
+    BinEdges edges(5, LinearGenerator(0, 2));
+    // Incorrect YMode
+    TS_ASSERT_THROWS(rebinFrequencies(getCountsHistogram(), edges),
```

Comment:
Don't need this, just provide a single `rebin()` that branches internally to call either `rebinCounts()` or `rebinFrequencies()`.